drivers/flash: stm32l4: Fix Coverity CID 203515.

LeiW000 opened the pull request:
It is complained in the coverity report that "pages_per_bank" is not be initialized.
So, we set it with a default value(DT_FLASH_SIZE >> 1).

Fix for #18361.

Signed-off-by: Steven Wang <steven.l.wang@linux.intel.com>

aescolar commented:
There is another fix queued 18478
This commit 0381e8c may have obsoleted the original Coverity bug

LeiW000 commented:
@aescolar Could you help to confirm if the issue has been fixed by 0381e8c? If no, Do you know who can help on it?

aescolar commented:
@LeiW000 No sorry, I cannot, I only helped triage it. Please inspect the code and see if the issue reported by Coverity is still relevant. Maybe @gon1332 or @erwango can help.

LeiW000 commented:
@aasthagr , Could you help to check the latest coverity report to see if the issue has been fixed with 0381e8c? I'm not sure it only by reviewing the code.

Commit:
drivers/flash: stm32l4: Fix Coverity CID 203515.

It is complained in the coverity report that "pages_per_bank"
is not be initialized. So, we set it with a default
value(DT_FLASH_SIZE >> 1).

Fix for #18361.

Signed-off-by: Steven Wang <steven.l.wang@linux.intel.com>

LeiW000 commented:
I'm closing this PR as I think commit 0381e8c can fix the issue.
